src/lib/parquet/src/Flow/Parquet/ParquetFile/RowGroupBuilder/ColumnChunkBuilder.php

```diff
@@ -43,15 +43,17 @@ private function createColumnChunkContainer(array $pageContainers, int $offset)
         $dictionaryPageSize = null;
         $dictionaryPageOffset = null;
         $pageOffset = $offset;
+        $dictionaryPage = null;
 
         foreach ($pageContainers as $pageContainer) {
             if ($pageContainer->pageHeader->type() === Type::DICTIONARY_PAGE) {
-                if ($dictionaryPageSize !== null) {
+                if ($dictionaryPage !== null) {
                     throw new RuntimeException('There can be only one dictionary page in column chunk');
                 }
 
                 $dictionaryPageOffset = $pageOffset;
                 $dictionaryPageSize = $pageContainer->size();
+                $dictionaryPage = $pageContainer;
```

Comment:
Can be simpler probably without `$dictionaryPage`:
```suggestion
                $valuesCount = \count($dictionaryPage->values);
```